### PR TITLE
jenkins: add options

### PR DESCRIPTION
**Pull request by lamping7**
- set timeout to 1 hour
- throw away old build logs, keep 10
- remove unnecessary pipeline agent, blocking the node
- formatting changes for readability

### CODE REVIEW

**Comment by lamping7** [comment]
@julianoes I threw this on top of #558

File: Jenkinsfile [hunk 1]
```diff
@@ -1,8 +1,13 @@
+#!/usr/bin/env groovy
```
Comment (julianoes) [inline Jenkinsfile]: ah!

File: Jenkinsfile [hunk 2]
```diff
@@ -123,10 +135,13 @@ pipeline {
             sh 'build/default/backend/test/unit_tests_backend'
           }
         }
-      }
-    }
+
+      } // parallel
+    } // Build
```
Comment (julianoes) [inline Jenkinsfile]: :+1:

File: Jenkinsfile [hunk 3]
```diff
@@ -196,6 +216,12 @@ pipeline {
...
         sh './generate_docs.sh'
       }
-    }
+    } // stage Generate Docs
+
+  } // stages
+
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10', artifactDaysToKeepStr: '30'))
+    timeout(time: 60, unit: 'MINUTES')
```
Comment (julianoes) [inline Jenkinsfile]: Makes sense!

**Pull request merged**